runtime/src/iree-amd-aie/aie_runtime/Utils/test/ChannelBdIdGeneratorTest.cpp

```diff
@@ -102,10 +93,26 @@ TEST(ChannelBdIdGeneratorTest, Release) {
   EXPECT_EQ(generator.isBdIdAssigned(1), true);
 }
 
+TEST(ChannelBdIdGeneratorTest, IncrementalAssign) {
```

Comment:
Add a couple of test as well for:
- Checking correctness when the max value is overflown (wrap).
- Checking correctness when all BDs are assigned/in use.